services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/PageableQuery.java

```diff
@@ -66,6 +68,25 @@ default <T> Page<T> findPage(
         return new PageImpl<>(results, pageable, aggregationResult.getMetadata().getFirst().getCount());
     }
 
+    default <T> Page<T> findGeneric(
+            @Nonnull MongoTemplate mongoTemplate,
+            @Nonnull String collectionName,
+            @Nonnull int limit,
+            @Nonnull Criteria criteria,
+            @Nonnull Sort sort,
+            @Nullable List<String> excludedFields,
+            @Nonnull Class<T> outputType) {
+        List<String> fieldsToExclude = excludedFields != null ? excludedFields : Collections.emptyList();
+        Query mongoQuery = Query.query(criteria).with(sort).limit(limit);
+        if (!fieldsToExclude.isEmpty()) {
+            for (String field : fieldsToExclude) {
+                mongoQuery.fields().exclude(field);
+            }
+        }
+        List<T> results = mongoTemplate.find(mongoQuery, outputType, collectionName);
+        return new PageImpl<>(results, PageRequest.of(0, limit), results.size());
+    }
+
```

Comment:
suggestion(blocking): Use the pageable interface instead of trying to roll our own implementations

```suggestion
    default <T> Page<T> findGeneric(
            @Nonnull MongoTemplate mongoTemplate,
            @Nonnull String collectionName,
            @Nonnull Criteria criteria,
            @Nullable List<String> excludedFields,
            @Nonnull Class<T> outputType,
            @Nonnull Pageable pageable) {
        List<String> fieldsToExclude = excludedFields != null ? excludedFields : Collections.emptyList();
        Query mongoQuery = Query.query(criteria).with(pageable.getSort()).limit(pageable.getPageSize());
        if (!fieldsToExclude.isEmpty()) {
            for (String field : fieldsToExclude) {
                mongoQuery.fields().exclude(field);
            }
        }
        List<T> results = mongoTemplate.find(mongoQuery, outputType, collectionName);
        return new PageImpl<>(results);
    }
```